src/main/java/cz/codecamp/logger/loggers/FunctionalMultiLogger.java

```diff
@@ -0,0 +1,23 @@
+package cz.codecamp.logger.loggers;
+
+import cz.codecamp.logger.LogLevelEnum;
+import cz.codecamp.logger.LoggerInterface;
+
+/**
+ * Created by honzapua on 9.10.2016.
+ */
+public class FunctionalMultiLogger extends MultiLogger {
+
+    public FunctionalMultiLogger(LoggerInterface... loggerInterfaces) { //zpropaguje konstruktor predka
+        super(loggerInterfaces);
+    }
+
+    @Override
+    public void log(LogLevelEnum level, String message) {
+        for (LoggerInterface loggerInterface : loggerInterfaces){
```

Comment:
`for` loop s dvojtečkou je stále imperativní způsob. Jako "funkcionální" by se dá nazvat [`Iterable.forEach`](https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html#forEach-java.util.function.Consumer-) s closures.